test/log_buffer/decorator_test.rb

```diff
@@ -15,6 +16,29 @@ def setup
     @decorator = ::Proxy::LogBuffer::Decorator.new(@logger, @buffer)
   end
 
+  def test_should_pass_logs
+    @logger.expects(:add).with(DEBUG, "message")
+    @decorator.debug("message")
+  end
+
+  def test_should_pass_logs_to_syslog
```

Comment:
I test the `add` method in case we will change it in future sending some methods to loggers which are not available in syslog implementation. This implementation is a bit unusual and missing some features (for obvious reasons). Therefore this test helps to avoid these mistakes (the patch actually fixes one of these - I was assuming that progname is a method that is implemented for all logger implementations).